RecoDataProducts/inc/CrvDAQerror.hh

```diff
@@ -0,0 +1,38 @@
+#ifndef RecoDataProducts_CrvDAQerror_hh
+#define RecoDataProducts_CrvDAQerror_hh
+//
+//
+// Contact person Ralf Ehrlich
+//
+
+#include <vector>
+
+namespace mu2e
+{
+  class CrvDAQerror
+  {
+    public:
+
+    enum errorCode{unableToGetDataBlock=0, invalidPacket=1, wrongSubsystemID=2, errorUnpackingStatusPacket=3, errorUnpackingCrvHits=4};
+
+    CrvDAQerror() {}
+
+    CrvDAQerror(int errorCode, int subEvent, int dataBlock, int packetCount) :
+               _errorCode(errorCode), _subEvent(subEvent), _dataBlock(dataBlock), _packetCount(packetCount) {}
+
+    int  GetErrorCode() const     {return _errorCode;}
+    int  GetSubEvent() const      {return _subEvent;}
+    int  GetDataBlock() const     {return _dataBlock;}
+    int  GetPacketCount() const   {return _packetCount;}
+
+    private:
+
+    int    _errorCode{0};
```

Comment:
How do you propose to deal with interpreting these values?
Could you use https://github.com/Mu2e/Offline/blob/b4e9f4fd506f1f4d57b4fdbca2184eb93e468eac/GeneralUtilities/inc/EnumToStringSparse.hh
to give values names?

Reply:
Yes, it could be done this way. Otherwise, we can just check the enums/ints to see if/which errors occurred.

Reply:
The strings are useful to communicate with humans (print).

Reply:
I updated this class using EnumToStringSparse.